script.js

```diff
@@ -0,0 +1,159 @@
+    {code: 'KeyB', en: 'b', enShift: 'B', ru: 'и', ruShift: 'И'},
+    {code: 'KeyN', en: 'n', enShift: 'N', ru: 'т', ruShift: 'Т'},
+    {code: 'KeyM', en: 'm', enShift: 'M', ru: 'ь', ruShift: 'Ь'},
+    {code: 'Comma', en: ',', enShift: '<', ru: 'б', ruShift: 'Б'},
+    {code: 'Period', en: '.', enShift: '>', ru: 'ю', ruShift: 'Ю'},
+    {code: 'Slash', en: '/', enShift: '?', ru: '.', ruShift: ','},
+    {code: 'ArrowUp', en: '↑', enShift: '↑', ru: '↑', ruShift: '↑'},
+    {code: 'ShiftRight', specKey: true, name: 'Shift'},
+
+    {code: 'ControlLeft', specKey: true, name: 'Ctrl'},
+    {code: 'MetaLeft', specKey: true, name: 'Win'},
+    {code: 'AltLeft', specKey: true, name: 'Alt'},
+    {code: 'Space', specKey: true, name: ''},
+    {code: 'AltRight', specKey: true, name: 'Alt'},
+    {code: 'ArrowLeft', en: '←', enShift: '←', ru: '←', ruShift: '←'},
+    {code: 'ArrowDown', en: '↓', enShift: '↓', ru: '↓', ruShift: '↓'},
+    {code: 'ArrowRight', en: '→', enShift: '→', ru: '→', ruShift: '→'},
+    {code: 'ControlRight', specKey: true, name: 'Ctrl'}
+  ];
+        
+  h1 = document.createElement('h1');
+  h1.innerHTML = 'Виртуальная клавиатура';
+  document.body.append(h1);
+
+  let textarea = document.createElement('textarea');
+  textarea.id = 'textarea';
+  textarea.autofocus = true;
+  document.body.append(textarea);
+
+  let keyboard = document.createElement('div');
+  keyboard.id = 'keyboard';
+  document.body.append(keyboard);
+
+  let addKeysDom = function (objKey) {
+    let key = document.createElement('div');
+    key.className = objKey.code;
+    key.innerText = objKey.en || objKey.name || objKey.code;
+    return document.getElementById('keyboard').append(key);
+  }
+
+  keys.forEach(objKey => addKeysDom(objKey));
+
+
+  class Button {
+
+    constructor(options) {
+      
+      this.code = options.code;
+      this.en = options.en;
+      this.enShift = options.enShift;
+      this.ru = options.ru;
+      this.ruShift = options.ruShift;
+      
+    }
+    
+  
+    pushKey (event) {
+      let localObjKey = {}
+      keys.forEach(function (obj) {
+       
+        if (obj.code === event.code ) {
+          let btn = document.querySelector(`.${obj.code}`);
+          localObjKey = obj;
+  
+          if (event.type === 'keydown') {
+            btn.classList.add('active');
+          } 
+          
+          if (event.type === 'keyup') {
+            btn.classList.remove('active');
+          }
+        }
+      });
+      
+      if(!localObjKey.specKey && event.type === 'keyup') {
+        document.getElementById('textarea').append(localObjKey.en);
+      } 
+    }
+
+    mouseClick(event) {
+    
+      keys.forEach(function (obj) {
+       
+        if(event.target.className === obj.code && !obj.specKey){
+          document.getElementById('textarea').append(obj.en);
+        }
+      });
+    }
+
+  }
+  
+  let button = new Button(keys);
+
+ document.getElementById('keyboard').addEventListener("click", (event) => {
```

Comment:
Use simplier notation for such a code:
`document.getElementById('keyboard').addEventListener("click", event => button.mouseClick(event));`

Use `querySelector()` or `querySelectorAll()` methods instead of `getElementById()`, `getElementByClassName()` or any similar methods. `querySelector()` and `querySelectorAll()` are more common and powerful methods.